### PR TITLE
Add validation for cloud provider Secrets

### LINKED ISSUE #90: Validate cloudprovider credentials
(recreating issue from the g/g repo: https://github.com/gardener/gardener/issues/2293)

**What would you like to be added**:
Add validation for `cloudprovider` secret

**Why is this needed**:
Currently, when uploading secrets via the UI, all secret fields are required and validated. However, when creating those credentials via the `cloudprovider` secret, there is no validation. This results in errors such as this error: (specific to Azure but a similar error would be generated for AWS):

```
Flow "Shoot cluster reconciliation" encountered task errors: [task "Waiting until shoot infrastructure has been reconciled" failed: failed to create infrastructure: retry failed with context deadline exceeded, last error: extension encountered error during reconciliation: Error reconciling infrastructure: secret shoot--xxxx--xxxx/cloudprovider doesn't have a subscription ID] Operation will be retried.
```

Screen Shot 2020-05-07 at 1 09 03 PM

### PR DESCRIPTION
**Pull request by vpnachev**
**How to categorize this PR?**
/area robustness
/area ops-productivity
/kind enhancement
/platform aws

**What this PR does / why we need it**:
Add validation for the aws access keys similar to https://github.com/gardener/gardener-extension-provider-gcp/pull/112.

**Which issue(s) this PR fixes**:
Fixes #90 
Ref: https://github.com/gardener/gardener/issues/2293

**Special notes for your reviewer**:

**Release note**:

```feature operator
The existing ValidatingWebhookConfiguration of `admission-aws` for Shoot validation does now validate also the Shoot secret. `admission-aws` does now feature also a new webhook that prevents Shoot secret to be updated with invalid keys.
```

### CODE REVIEW

File: pkg/apis/aws/validation/secrets.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	if len(accessKeyID) > accessKeyMaxLen {
+		return fmt.Errorf("field %q in secret %s cannot be longer than %d characters", aws.AccessKeyID, secretRef, accessKeyMaxLen)
+	}
+
+	secretAccessKey, ok := secret.Data[aws.SecretAccessKey]
+	if !ok {
+		return fmt.Errorf("missing %q field in secret %s", aws.SecretAccessKey, secretRef)
+	}
+
+	if len(secretAccessKey) < secretAccessKeyMinLen {
```
Comment (dkistner) [inline pkg/apis/aws/validation/secrets.go:52]: Is there no max length for the `secretAccessKey` as well?

**Review by dkistner — DISMISSED** [review]
Very nice.

/lgtm
/invite @kon-angelo

File: pkg/apis/aws/validation/secrets.go (hunk 1 above)
Comment (vpnachev) [reply pkg/apis/aws/validation/secrets.go:52]: I tried to quickly find some documentation about this, but couldn't. Then decided to implement what the dashboard is doing - https://github.com/gardener/dashboard/blob/6675a7194d95ef1e4b5b22fde86c9ee306b86c88/frontend/src/components/dialogs/SecretDialogAws.vue#L182-L188

Comment (dkistner) [reply pkg/apis/aws/validation/secrets.go:52]: I also quickly checked and did not find anything in the aws docs. Thank you

**Review by stoyanr — APPROVED** [review]
/lgtm

File: pkg/apis/aws/validation/secrets.go (hunk 1 above)
Comment (kon-angelo) [reply pkg/apis/aws/validation/secrets.go:52]: Take this with a grain of salt, but the official documentation does not mention any pattern. On one hand, the vue snipper is about 3 years old with no change (so quite stable), on the other hand I didn't find any similar (min or max) check in the go code.

File: pkg/apis/aws/validation/secrets.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+// ValidateCloudProviderSecret checks whether the given secret contains a valid AWS access keys.
+func ValidateCloudProviderSecret(secret *corev1.Secret) error {
+	secretRef := fmt.Sprintf("%s/%s", secret.Namespace, secret.Name)
+
+	accessKeyID, ok := secret.Data[aws.AccessKeyID]
+	if !ok {
+		return fmt.Errorf("missing %q field in secret %s", aws.AccessKeyID, secretRef)
+	}
+
+	if len(accessKeyID) < accessKeyMinLen {
```
Comment (kon-angelo) [inline pkg/apis/aws/validation/secrets.go:39]: Shouldn't this be inclusive checks ?

**Review by kon-angelo — APPROVED** [review]
/lgtm

**Pull request merged**